### PR TITLE
clang: Fix emulated TLS for libstdc++ envs

### PR DESCRIPTION
**Pull request by lazka**
See https://github.com/msys2/MINGW-packages/pull/12550#issuecomment-1228506687
and https://github.com/msys2/msys2-tests/pull/9

### CODE REVIEW

File: mingw-w64-clang/PKGBUILD [hunk 1]
```diff
@@ -67,7 +67,8 @@ source=("${_url}/llvm-${pkgver}.src.tar.xz"{,.sig}
         "0301-LLD-COFF-Add-support-for-a-new-mingw-specific-embedd.patch"
         "0302-LLD-MinGW-Implement-the-exclude-symbols-option.patch"
         "0303-ignore-new-bfd-options.patch"
-        "0401-Do-not-try-to-build-CTTestTidyModule-for-Windows-with-dylibs.patch")
+        "0401-Do-not-try-to-build-CTTestTidyModule-for-Windows-with-dylibs.patch"
+        "0500-fix-emulated-tls.patch")
```
Comment (mati865) [inline mingw-w64-clang/PKGBUILD:71]: What does `0500` relate to?

File: mingw-w64-clang/PKGBUILD [hunk 2]
```diff
@@ -99,7 +100,8 @@ sha256sums=('050922ecaaca5781fdf6631ea92bc715183f202f9d2f15147226f023414f619a'
             'e6c3b97f733920b0a32e889cd180dd86507bddea09e9c217cd1445c047cd4976'
             'bd40b6c84d25de5ac5877f2c83ac61d872838bc034df62e98ad16116d23744e0'
             '778e0db0a5b0657ab05e2a81d83241347a4a41af2b0f9903422f651fa58e8d40'
-            'a10e5192889501c75db6a41e29df44b2af8502b3ef34c2315789563a5c8e9b85')
+            'a10e5192889501c75db6a41e29df44b2af8502b3ef34c2315789563a5c8e9b85'
+            '627739ff0befaa0851bc991f75f912507e667a0ceabc18ddd0422694c56fb636')
```
Comment (mati865) [inline mingw-w64-clang/PKGBUILD:104]:
```suggestion
            '3a58bd5243fb082df64c90bd9a4f00c698317f20365d11bbe30fac5dfccd215e')
```

File: mingw-w64-clang/PKGBUILD (hunk 1 above)
Comment (MehdiChinoune) [reply mingw-w64-clang/PKGBUILD:71]: It should between 0001->0099

Comment (mati865) [reply mingw-w64-clang/PKGBUILD:71]: Actually 0101-0199 because it's for Clang.

Comment (MehdiChinoune) [reply mingw-w64-clang/PKGBUILD:71]: yes.

Comment (lazka) [reply mingw-w64-clang/PKGBUILD]: ah, sorry.

**Comment by mstorsjo** [comment]
FYI, just a headsup regarding this:

I tried to upstream this patch as https://reviews.llvm.org/D132848, but a different solution was requested in the end, reviewed as https://reviews.llvm.org/D132916 and finally landed as https://github.com/llvm/llvm-project/commit/fbfe1db4a95a73ed6a0767db0ab7d449fc03405e. The tricky thing is that it probably won't give a merge conflict when you rebase your patches on top of a version that carries this commit, so you might end up carrying the downstream patch longer than necessary if you don't notice this (in e.g. the 16.x release).

(Switching and backporting https://github.com/llvm/llvm-project/commit/fbfe1db4a95a73ed6a0767db0ab7d449fc03405e to the msys2 patched version instead of this one might also be feasible.)

**Comment by lazka** [comment]
> FYI, just a headsup regarding this:

thanks, part of #13253

**Pull request merged**